FirebaseDatabase/Sources/Utilities/FNextPushId.m

```diff
@@ -93,19 +100,91 @@ + (NSString *)successor:(NSString *_Nonnull)key {
         return [FUtilities maxName];
     }
 
-    NSString *source =
-        [NSString stringWithFormat:@"%C", [next characterAtIndex:i]];
-    NSInteger sourceIndex = [PUSH_CHARS rangeOfString:source].location;
-    NSString *sourcePlusOne = [NSString
-        stringWithFormat:@"%C", [PUSH_CHARS characterAtIndex:sourceIndex + 1]];
+    unichar character = [next characterAtIndex:i];
+    unichar plusOne = character + 1;
+    BOOL removePreviousCharacter = NO;
+    BOOL replaceWithLowestSurrogatePair = NO;
+    switch (plusOne) {
+    case 0x23: // '#'
+    case 0x24: // '$'
+        plusOne = 0x25;
+        break;
```

Comment:
Can we do:

```suggestion
    case '#':
    case '$':
        plusOne ='.';
        break;
```

Reply:
Yes, we can certainly do that. My reasoning for using numerical values is that I think that it is easier to follow that we are adding or skipping or what we are doing. In the 'ordering' sense, the actual character value doesn't really matter as I see it.

But let me know if you think I should change it.

Reply:
I didn't think about it like that - let's leave it as is. The comments do help with the overall understanding.